pytorch_lightning/utilities/model_summary.py

```diff
@@ -85,19 +85,20 @@ def _register_hook(self) -> Optional[RemovableHandle]:
             A handle for the installed hook, or ``None`` if registering the hook is not possible.
         """
 
-        def hook(module, inp, out):
+        def hook(_: nn.Module, inp: Any, out: Any) -> None:
             if len(inp) == 1:
                 inp = inp[0]
             self._in_size = parse_batch_shape(inp)
             self._out_size = parse_batch_shape(out)
+            assert self._hook_handle is not None
```

Comment:
reviewer: the outer function here registers the hook, so the handle must not be None. and if it was, it would mean the hook would not run. so this assert is good.